include/nuttx/streams.h

```diff
@@ -217,6 +217,14 @@ struct lib_rawsostream_s
   int                    fd;
 };
 
+struct lib_bufferedoutstream_s
+{
+  struct lib_outstream_s public;
+  int                    fd;
```

Comment:
let's change fd to:
FAR struct lib_outstream_s *backend;
so the buffer can apply to all stream

Reply:
Done